src/main/java/org/folio/service/processing/ranking/ScoreService.java

```diff
@@ -0,0 +1,140 @@
+package org.folio.service.processing.ranking;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NavigableSet;
+import java.util.Optional;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dao.DataImportQueueItemDao;
+import org.folio.rest.jaxrs.model.DataImportQueueItem;
+import org.folio.rest.jaxrs.model.DataImportQueueItemCollection;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ScoreService {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  @Autowired
+  private QueueItemHolisticRanker ranker;
+
+  @Autowired
+  private DataImportQueueItemDao queueItemDao;
+
+  /**
+   * Get the next best queue item that is currently waiting to execute
+   */
+  public Future<Optional<DataImportQueueItem>> getBestQueueItem() {
+    return this.getRankedQueueItems()
+      .map((NavigableSet<DataImportQueueItem> set) -> {
+        if (set.isEmpty()) {
+          return Optional.empty();
+        }
+        return Optional.of(set.first());
+      });
+  }
+
+  /**
+   * Get a list of all WAITING queue items, sorted by score, with the highest
+   * score coming first
+   */
+  public Future<NavigableSet<DataImportQueueItem>> getRankedQueueItems() {
+    // we create a temporary map to store these, since the sorting comparator may
+    // require multiple calls to `calculateScore`.
+    // this is local to this method, since we don't want to cache these any longer
+    // than one request/sorting.
+    Map<DataImportQueueItem, Double> cache = new HashMap<>();
+
+    return CompositeFuture
+      .all(
+        queueItemDao.getAllInProgressQueueItems(),
+        queueItemDao.getAllWaitingQueueItems()
+      )
+      .map((CompositeFuture result) -> {
+        DataImportQueueItemCollection inProgress = result.resultAt(0);
```

Comment:
I did, but since we're going to have multiple workers, we'd need to perform queries anyways to determine what jobs have been taken since, which tenants are using most resources, etc.; two `SELECT` calls is a very small performance hit IMO.